### ISSUE TITLE
Cross-Compile for aarch64

### ISSUE THREAD

**Issue by MoDa89** [go-gst/go-gst#100]
Hi,

I'm trying to cross-compile the binding for aarch64-target, but it gives an error from gcc:

# github.com/go-gst/go-glib/glib
aarch64-none-linux-gnu-gcc: ERROR: unsafe header/library path used in cross-compilation: '-I/usr/include/libmount'

it shows that it's attempting to include headers from building machine, not the aarch64 target. 
Even though there are other CGO parts in the project that are compiling well (pointing to the target's rootfs container),
I can't get this part to do so.

Any help is appreciated.

**Comment by RSWilli** [comment]
Heads-up: I have never tried to cross compile the library, and I suspect that there are a lot of headaches involved.

the glib and gst bindings use pkgconfig to discover the include paths: https://github.com/go-gst/go-glib/blob/main/glib/glib.go#L21, I suspect that this adds the undesired include paths for you.

**Comment by MoDa89** [comment]
Yes, you're right the issue was in the cgo pkg-config command, and yes it was a headache to pinpoint.

In short, I had to invoke the flag "--define-prefix" which basically "guesstimates" (to quote the man-page)
the relevant libraries paths based on the relative .pc files paths in the target-container , which over-writes 
the original cflags/libs path, and thus stops gcc from leaking the host-machine's cflags/libs paths into 
the target-container's. Of course, this was accompanied by the headache of dragging the whole 
dependency-tree along with, as below (from glib/giostream.go):

[original]
#cgo pkg-config: gio-2.0 glib-2.0 gobject-2.0

[modified]
#cgo pkg-config: --print-errors --define-prefix gio-2.0 glib-2.0 gobject-2.0 gmodule-2.0 gmodule-no-export-2.0 zlib mount libffi libpcre2-8 blkid

Good thing it worked and now I can run it on ARM64 targets. Thanks a lot.

**Comment by RSWilli** [comment]
Another approach could also be to set the `PKG_CONFIG_PATH` environment variable. The following article looks promising: https://dh1tw.de/2019/12/cross-compiling-golang-cgo-projects/

**Comment by MoDa89** [comment]
I don't think there's any problem in placing the modified "pkg-config" command instead of the original, as in any case it would either over-write the cflags/libs paths with the designated ones (the build container, if any), or keep them unchanged (e.g. in case the target-machine arch. is the same as the building-machine arch.)  

As for the PKG_CONFIG_PATH, I have tried that (pretty similar to the implementation in attached link), but gcc didn't pick up on that envar for each required dependency, as it turned out the error wasn't about the glib/gobject dependencies (first-degree dependencies, per se), but rather reached out to the third-degree dependencies (e.g. blkid, libffi and mount libraries, which was the core issue, in the first place), in which I had to investigate the dependencies in every .pc file for each required dependency and append it to pkg-config's "--define-prefix" for over-write.  Good thing the dependency-tree didn't turn out to be that big.

**Comment by RSWilli** [comment]
closing because stale, old, or redundant
